packages/@lwc/integration-karma/test/component/refs/index.spec.js

```diff
@@ -178,15 +179,24 @@ describe('refs', () => {
         expect(elm.getRefTextContent('foo')).toEqual('foo');
     });
 
-    it('ref on a dynamic component', () => {
-        const elm = createElement('x-dynamic', { is: Dynamic });
+    it('ref on a dynamic component - lwc:dynamic', () => {
+        const elm = createElement('x-dynamic', { is: LwcDynamic });
         document.body.appendChild(elm);
 
         const dynamic = elm.getRef('dynamic');
         expect(dynamic.tagName.toLowerCase()).toEqual('x-dynamic-cmp');
         expect(dynamic.getRefTextContent('first')).toEqual('first');
     });
 
+    it('ref on a dynamic component - <lwc:component lwc:is={}>', () => {
+        const elm = createElement('x-dynamic', { is: Dynamic });
+        document.body.appendChild(elm);
+
+        const dynamic = elm.getRef('dynamic');
+        expect(dynamic.tagName.toLowerCase()).toEqual('x-basic');
+        expect(dynamic.getRefTextContent('first')).toEqual('first');
```

Comment:
`this.refs.dynamic` is undefined if the `ctor` is undefined/null, correct? Should we have a test for that?

Reply:
Updated these [tests](https://github.com/salesforce/lwc/blob/1658d36d1a001bfce3a4a441323e9d4caf473f48/packages/%40lwc/integration-karma/test/component/refs/index.spec.js#L182-L216) to check for the `ref` when the constructor is undefined as well.